File: 프로그래머스/정렬/가장큰수.cpp
//https://sohyunwriter.tistory.com/140

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string a, string b){
    return a + b > b + a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> v;
    
    for(auto i:numbers){
        v.push_back(to_string(i));
    }
    sort(v.begin(), v.end(), compare);
    if(v[0] == "0"){ // {0, 0, 0, 0} -> "0"
        return "0";
    }
    for(auto i : v)
        answer += i;
    return answer;
}

File: 프로그래머스/정렬/가장 큰 수.py
import functools

def mycmp(a, b):
    if (b+a) < (a+b):
        return -1
    else:
        return 1

def solution(numbers):
    s_nums = list(map(str, numbers))
    s_nums.sort(key=functools.cmp_to_key(mycmp))
    answer = str(int(''.join(s_nums)))
        
    return answer
